### PR TITLE
feat: implement imperative api for android picker

### PR DESCRIPTION
**Pull request by vonovak**
# Summary

On Android, this gives a choice between using the component API (regular React component) or an imperative api (think something like ReactNative.alert()).

The component API has the benefit of writing the same code on all platforms, but on Android, the date/time picker opens in a dialog, similar to ReactNative.alert() from core react native. The imperative api models this behavior better than the declarative component api. While the component approach is perfectly functional, based on the issue tracker history, it appears to be more prone to introducing bugs.

## Test Plan

The component api internally uses the newly added imperative one, and so the e2e tests which are written using the component api also cover the new imperative one

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |     ❌     |
| Android |    ✅      |

## Checklist

- [x] I have tested this on a device and a simulator
- [x] I added the documentation in `README.md`
- [x] I updated the typed files (TS and Flow)
- [x] I added a sample use of the API in the example project (`example/App.js`)
- [x] I have added automated tests, either in JS or e2e tests, as applicable

### CODE REVIEW

**Comment by vonovak** [comment]
:tada: This issue has been resolved in version 6.0.0 :tada:

If this package helps you, consider [sponsoring us](https://github.com/sponsors/react-native-datetimepicker)! :rocket:

File: src/DateTimePickerAndroid.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+          break;
+      }
+    } catch (error) {
+      onError && onError(error);
+    }
+  };
+  presentPicker();
+}
+
+function dismiss(mode: AndroidNativeProps['mode']) {
```
Comment (flochtililoch) [inline src/DateTimePickerAndroid.js:97]: This syntax seem to be invalid with older versions of RN/flow (tested on RN 63.3/flow 0.122.0):

```
error: SyntaxError: .../node_modules/@react-native-community/datetimepicker/src/DateTimePickerAndroid.js: Unexpected token, expected "]" (97:42)

   95 | }
   96 |
>  97 | function dismiss(mode: AndroidNativeProps['mode']) {
      |                                           ^
   98 |   // $FlowFixMe - `AbstractComponent` [1] is not an instance type.
   99 |   pickers[mode].dismiss();
  100 | }
```

Comment (vonovak) [reply src/DateTimePickerAndroid.js:97]: hello, yes, you'll need Flow 155: https://flow.org/en/docs/types/indexed-access/

Comment (sedatbasar) [reply src/DateTimePickerAndroid.js:97]: We are not using flow in project RN 0.63.4 and having this issue. Not sure how to fix. Do you have a workaround?
Tried to use older versions which is 5.1.0 and 3.5.2 but for those i am getting "Unhandled promise rejection [TypeError: Cannot read property 'open' of null" any idea or guidence to help on this?

**Comment by TechWithMuskan** [comment]
@flochtililoch Have you found any solution of this error, experiencing same
error: SyntaxError: /Users/Documents/MyFirstApp/node_modules/@react-native-community/datetimepicker/src/DateTimePickerAndroid.android.js: Unexpected token, expected "]"

**Comment by sedatbasar** [comment]
> @flochtililoch Have you found any solution of this error, experiencing same error: SyntaxError: /Users/Documents/MyFirstApp/node_modules/@react-native-community/datetimepicker/src/DateTimePickerAndroid.android.js: Unexpected token, expected "]"

Used older versions of it.

**Pull request merged**